src/views/withNavigationFocus.js

```diff
@@ -0,0 +1,56 @@
+import React from 'react';
+import propTypes from 'prop-types';
+import hoistStatics from 'hoist-non-react-statics';
+import invariant from '../utils/invariant';
+
+export default function withNavigationFocus(Component) {
+  class ComponentWithNavigationFocus extends React.Component {
+    static displayName = `withNavigationFocus(${Component.displayName ||
+      Component.name})`;
+
+    static contextTypes = {
+      navigation: propTypes.object.isRequired,
+    };
+
+    state = {
+      isFocused: false,
```

Comment:
as per my most recent comment, if we exposed `isFocused()` then this would just be:

```
constructor(props, context) {
  super();
  const navigator = props.navigator || context.navigator;
  this.state = {
    isFocused: navigation.isFocused(),
  };
}
```

Reply:
So `isFocused` is a render-unsafe fn that we would use only to init HOC state? Couldn't this lead to users trying to using it directly and being confused when no re-renders happen? Would this function be documented for general usage?

Reply:
i think if they use it in a function and are confused by no re-renders happening then they have misunderstood how react works, that's not really a problem of the library

Reply:
that's true, and it still can be advised to use the HOC anyway.
I'll try to check soon how to provide that `isFocused` func